source/common/config/new_grpc_mux_impl.cc

```diff
@@ -93,32 +95,17 @@ void NewGrpcMuxImpl::onEstablishmentFailure() {
   } while (all_subscribed.size() != subscriptions_.size());
 }
 
-void NewGrpcMuxImpl::onWriteable() { trySendDiscoveryRequests(); }
-
-void NewGrpcMuxImpl::kickOffAck(UpdateAck ack) {
-  pausable_ack_queue_.push(std::move(ack));
-  trySendDiscoveryRequests();
-}
-
-// TODO(fredlas) to be removed from the GrpcMux interface very soon.
-GrpcMuxWatchPtr NewGrpcMuxImpl::subscribe(const std::string&, const std::set<std::string>&,
-                                          GrpcMuxCallbacks&) {
-  NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
-}
-
-void NewGrpcMuxImpl::start() { grpc_stream_.establishNewStream(); }
-
 Watch* NewGrpcMuxImpl::addWatch(const std::string& type_url, const std::set<std::string>& resources,
                                 SubscriptionCallbacks& callbacks,
                                 std::chrono::milliseconds init_fetch_timeout) {
-  auto entry = subscriptions_.find(type_url);
-  if (entry == subscriptions_.end()) {
+  auto watch_map = watch_maps_.find(type_url);
+  if (watch_map == watch_maps_.end()) {
     // We don't yet have a subscription for type_url! Make one!
     addSubscription(type_url, init_fetch_timeout);
     return addWatch(type_url, resources, callbacks, init_fetch_timeout);
```

Comment:
Can we save the recursive?

Reply:
Yeah, fair enough, the recursion is a little "cute". Done.